packages/styled-components/components/text/index.js

```diff
@@ -1,53 +1,113 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import sanitizeHtml from 'sanitize-html';
+import EmbedContainer from 'react-oembed-container';
+import { richText } from '@irvingjs/core/config/html';
 import withThemes from '@irvingjs/styled/components/withThemes';
 import * as defaultStyles from './themes/default';
+import * as htmlStyles from './themes/html';
+import * as unstyledStyles from './themes/unstyled';
+import * as captionStyles from './themes/caption';
+import * as h1Styles from './themes/h1';
+import * as h2Styles from './themes/h2';
+import * as h3Styles from './themes/h3';
+import * as h4Styles from './themes/h4';
+import * as h5Styles from './themes/h5';
+import * as h6Styles from './themes/h6';
 
 /**
- * Render children in a React fragment or any other HTML tag.
+ * Output text.
  */
 const Fragment = (props) => {
   const {
-    children,
+    content,
+    html,
+    oembed,
+    style,
     tag,
-    theme,
+    theme = defaultStyles,
   } = props;
 
-  const { Element } = theme;
-
-  return (
-    <>
-      {tag ? (
-        <Element as={tag}>{children}</Element>
-      ) : (
-        children
-      )}
-    </>
-  );
+  const { TextWrapper } = theme;
+
+  switch (true) {
+    case 0 === content.length:
+      return false;
+
+    case true === oembed:
+      return (
+        <EmbedContainer markup={content}>
+          <TextWrapper
+            as={tag}
+            dangerouslySetInnerHTML={{ __html: sanitizeHtml(content, richText) }} // eslint-disable-line react/no-danger, max-len
+            style={style}
+          />
+        </EmbedContainer>
+      );
+
+    case true === html:
+      return (
+        <TextWrapper
+          as={tag}
+          dangerouslySetInnerHTML={{ __html: sanitizeHtml(content, richText) }} // eslint-disable-line react/no-danger
+          style={style}
+        />
+      );
+
+    default:
+      return (
+        <TextWrapper as={tag} style={style}>{content}</TextWrapper>
+      );
+  }
 };
 
 Fragment.defaultProps = {
-  children: {},
-  tag: null,
+  content: '',
+  html: false,
+  oembed: false,
+  style: {},
+  tag: 'div',
+  theme: defaultStyles,
 };
 
 Fragment.propTypes = {
   /**
-   * Children of the component.
+   * Markup to render.
+   */
+  content: PropTypes.string,
+  /**
+   * Render as HTML.
+   */
+  html: PropTypes.bool,
+  /**
+   * Render oembeds.
+   */
+  oembed: PropTypes.bool,
+  /**
+   * CSS styles.
    */
-  children: PropTypes.node,
+  style: PropTypes.object,
   /**
-   * Tag used to render.
+   * Wrapping element.
    */
   tag: PropTypes.string,
   /**
    * Theme (styles) to apply to the component.
    */
-  theme: PropTypes.object.isRequired,
+  theme: PropTypes.object,
 };
 
 const themeMap = {
   default: defaultStyles,
+  html: htmlStyles,
+  unstyled: unstyledStyles,
+  caption: captionStyles,
+  h1: h1Styles,
+  h2: h2Styles,
+  h3: h3Styles,
+  h4: h4Styles,
+  h5: h5Styles,
+  h6: h6Styles,
```

Comment:
None of these are final. Just a simple first-pass before some frontend devs can make some better decisions around it.